juju/juju
# Add open telemetry agent config

This is the precursor work to enable open telemetry in jujud binaries.
We require that information upon start-up of an agent and the only
way currently to do that is via setting it in the agent config.

Most of this code copies the work I did for the query tracing in
dqlite. So it's relatively easy to wire it all up, it's just very long-winded.

## Checklist


## QA steps

```sh
$ juju bootstrap lxd test --build-agent
$ juju controller-config "open-telemetry-enabled=true"
$ juju controller-config "open-telemetry-endpoint=http://10.0.0.1:16686"
```

You should see the agent restart and those values set in the agent config.
